src/Exceptions/UnsupportedMethodException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace PrestaShop\CircuitBreaker\Exceptions;
+
+/**
+ * Used when trying to use an unsupported HTTP method
+ */
+class UnsupportedMethodException extends CircuitBreakerException
+{
+
+    /**
+     * @param string $methodName
+     *
+     * @return UnsupportedMethodException
+     */
+    public static function unsupportedMethod($methodName)
```

Comment:
we can do better : InvalidParameter::unsupportedMethod($method)

Reply:
That would prevent people from being able to catch that specific error.

Reply:
Could also be in constructor :thinking: 

```php
public function __construct($methodName) 
{
    parent::__construct(sprintf('Unsupported method: "%s"', $methodName));
}
```

Reply:
That constructor signature does not comply with the parent's.